content_sync/tasks.py

```diff
@@ -392,3 +394,119 @@ def check_incomplete_publish_build_statuses():
             log.exception(
                 "Error updating publishing status for website %s", website.name
             )
+
+
+@app.task(acks_late=True)
+def update_websites_in_root_website():
+    """
+    Get all websites published to draft / live at least once, and for each one create or update
+    a WebsiteContent object of type website in the website denoted by settings.ROOT_WEBSITE_NAME
+    """
+    if settings.CONTENT_SYNC_BACKEND:
+        root_website = Website.objects.get(name=settings.ROOT_WEBSITE_NAME)
+        # Get all sites, minus any sites that have never been successfully published
+        sites = Website.objects.exclude(
+            Q(**{"draft_publish_date__isnull": True})
+            & Q(**{"publish_date__isnull": True})
+        )
+        sites = sites.exclude(Q(url_path__isnull=True))
+        # Exclude the root website
+        sites = sites.exclude(name=settings.ROOT_WEBSITE_NAME)
+        fields = [
+            "website",
+            "type",
+            "title",
+            "dirpath",
+            "filename",
+            "is_page_content",
+            "metadata",
+        ]
+        with WebsiteContent.bulk_objects.bulk_update_or_create_context(
+            fields, match_field="filename", batch_size=100
+        ) as bulk_update:
+            for website in sites:
+                site_metadata = WebsiteContent.objects.get(
+                    website=website, type="sitemetadata"
+                ).metadata
+                # We want this content to show up in lists, but not render pages
+                site_metadata["_build"] = {"list": True, "render": False}
+                # Set the content to draft if the site has not been published or is unpublished
+                if website.unpublish_status is not None or website.publish_date is None:
+                    site_metadata["draft"] = True
+                # Carry over url_path for proper linking
+                site_metadata["url_path"] = website.url_path
+                bulk_update.queue(
+                    WebsiteContent(
+                        website=root_website,
+                        type="website",
+                        title=website.title,
+                        dirpath=WEBSITE_LISTING_DIRPATH,
+                        filename=website.short_id,
+                        is_page_content=True,
+                        metadata=site_metadata,
+                    )
+                )
+        website_content = WebsiteContent.objects.filter(
+            website=root_website, type="website"
+        )
+        backend = api.get_sync_backend(website=root_website)
+        backend.sync_all_content_to_backend(query_set=website_content)
```

Comment:
I ran `docker-compose run web python manage.py update_websites_in_root_website` after modifying a site's metadata (title), and the corresponding `website` ContentFile was updated in the database, but no new commit was made to git.  I suspect it's because the signal to update ContentSyncState might not have been triggered by the bulk update?